### PR TITLE
[SPARK-49234][BUILD] Upgrade `xz` to `1.10`

### PR DESCRIPTION
**Pull request by dongjoon-hyun**
### What changes were proposed in this pull request?

This PR aims to upgrade `xz` to `1.10` independently from Apache Avro change.

### Why are the changes needed?

`1.10` is the latest minor version update with new improvements (like `ARM64`, `optimized classes for Java >= 9`) and bug fixes.
- https://github.com/tukaani-project/xz-java/blob/master/NEWS.md#110-2024-07-29

Note that the license is also changed from `Public Domain` to `BSD Zero Clause`.
> Licensing change: From version 1.10 onwards, XZ for Java is under the BSD Zero Clause License (0BSD). 1.9 and older are in the public domain and obviously remain so; the change only affects the new releases.

### Does this PR introduce _any_ user-facing change?

No behavior change.

### How was this patch tested?

Pass the CIs.

### Was this patch authored or co-authored using generative AI tooling?

No.

### CODE REVIEW

**Comment by dongjoon-hyun** [comment]
Since you did `xz` `1.9` upgrade, WDYT, @LuciferYang ?
- #38538

File: LICENSE-binary [hunk 1]
```diff
@@ -422,6 +422,11 @@ Python Software Foundation License
 python/pyspark/loose_version.py
 
 
+BSD 0-Clause
```
Comment (LuciferYang) [inline LICENSE-binary:425]: Do we need to update the binary LICENSE file?

Comment (dongjoon-hyun) [reply LICENSE-binary:425]: Oh, right.

Comment (dongjoon-hyun) [reply LICENSE-binary:425]: `licenses/LICENSE-xz.txt` is added.

File: licenses/LICENSE-xz.txt [hunk 2]
```diff
@@ -0,0 +1,11 @@
+Permission to use, copy, modify, and/or distribute this
```
Comment (dongjoon-hyun) [inline licenses/LICENSE-xz.txt:1]: This comes from 
- https://github.com/tukaani-project/xz-java/blob/master/LICENSES/0BSD.txt

**Review by HyukjinKwon — APPROVED** [review]

**Review by LuciferYang — APPROVED** [review]
LGTM

**Comment by dongjoon-hyun** [comment]
Thank you, @HyukjinKwon and @LuciferYang .

**Review by yaooqinn — APPROVED** [review]

**Comment by dongjoon-hyun** [comment]
Thank you, @yaooqinn .

**Comment by LuciferYang** [comment]
Merged into master. Thanks @dongjoon-hyun @HyukjinKwon and @yaooqinn

**Comment by dongjoon-hyun** [comment]
Thank you again @LuciferYang .

File: licenses/LICENSE-xz.txt (hunk 2 above)
Comment (yaooqinn) [inline licenses/LICENSE-xz.txt:1]: Hi @dongjoon-hyun

This file should be placed in the licenses-binary folder as the `xz` belongs to the binary releases

Comment (dongjoon-hyun) [reply licenses/LICENSE-xz.txt:1]: Thank you. Let me make a follow-up, @yaooqinn .

Comment (dongjoon-hyun) [reply licenses/LICENSE-xz.txt:1]: Thank you, @yaooqinn .
Here is the follow-up.
- https://github.com/apache/spark/pull/48107

**Pull request closed without merging**